sql/src/parser.rs

```diff
@@ -818,6 +914,94 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_create_partition_table_with_hash() {
```

Comment:
This test is too large, spilt in some same tests?

Reply:
Done.